massgov/mayflower
# DP-5329 - adds browser history enabled pagination for listing pages

This PR adds a URL parameter indicating the page, enables the back/forward browser buttons, and preserves state upon reload of listing pages. 

**To Test**
_note: this will not work within patternlab with the toolbar present, go to the individual pages as listed below._
For pages http://localhost:3000/patterns/05-pages-location-listing/05-pages-location-listing.html and http://localhost:3000/patterns/05-pages-event-listing/05-pages-event-listing.html respectively...
1. Load page in browser, confirm that clicking between pages via numbered buttons or next/previous updates the URL to contain `page=X` without reloading the page. 
2. Confirm that after switching between pages, clicking the browser buttons 'back' and 'forward' moves you between pages
3. Open a new browser window and go _directly_ to `http://localhost:3000/patterns/05-pages-location-listing/05-pages-location-listing.html ?page=2` and confirm that the second page is loaded.
